src/ecs/resources/input.rs

```diff
@@ -333,9 +333,9 @@ impl InputHandler {
     }
 
     /// Returns the value of an axis by the i32 id, if the id doesn't exist this returns None.
-    pub fn axis_value(&self, id: i32) -> Option<f32> {
+    pub fn axis_value(&self, id: &str) -> Option<f32> {
```

Comment:
@Xaeroxe Sorry, perhaps I wasn't clear enough. I am suggesting we use `AsRef<str>`, not `Into<String>`. I had mistakenly pointed out that `AsRef<str>` would reduce unnecessary allocations, confusing it for `Into<String>`. However, I still stand by my point that `AsRef<str>` will behave identically to `&str` except it will allow users to pass in other string slice types like `OsStr` and `CStr` with less boilerplate.